docs: Update Google Sign-In setup and configuration documentation (#471)
serverpod/serverpod_docs: docs/06-concepts/11-authentication/04-providers/03-google/02-customizations.md

# Customizations

This page covers additional configuration options for the Google identity provider beyond the basic setup.

## Configuration options

Below is a non-exhaustive list of some of the most common configuration options. For more details on all options, check the `GoogleIdpConfig` in-code documentation.

Serverpod provides two configuration classes:

- **`GoogleIdpConfigFromPasswords`** -- Loads the client secret from `passwords.yaml` or the `SERVERPOD_PASSWORD_googleClientSecret` environment variable. This is the class used in the [setup guide](./01-setup.md) and is recommended for most projects.
- **`GoogleIdpConfig`** -- The base class that requires you to provide the client secret explicitly. Use this when you need custom control over how credentials are loaded.

Both classes accept the same optional callbacks shown in the sections below. The examples on this page use `GoogleIdpConfigFromPasswords` unless the section specifically demonstrates manual client secret loading.

> ### Loading Google Client Secret

Reviewer: This section is about the GoogleIdpConfig. Let's make it clear:

```suggestion
### Loading the client secret using GoogleIdpConfig
```

Also remember to sentence case headings.

> If you use `GoogleIdpConfig` instead of `GoogleIdpConfigFromPasswords`, you must provide the client secret explicitly. You can load it in several ways:

Reviewer: Mentioning both GoogleIdpConfig and GoogleIdpConfigFromPasswords here makes it confusing. We can simplify this:

```suggestion
When using `GoogleIdpConfig`, you must provide the client secret explicitly. 

You can load the secret in several ways:
```

Reviewer: Also: 

1. What does "providing the secret explicitly" mean? Can we clarify that?
2. Are these the only three ways to do this or are there several more ways? Feels like something is missing here. What are the different ways? And what is common use cases?

Author: Updated! The intro now clarifies what "providing the secret explicitly" means (passing a `GoogleClientSecret` object directly) and explains that these are the only two configuration classes. Also added context on when you'd use each one.

For the "Loading Google Client Secret" section: the three examples (`fromJsonString`, `fromJsonFile`, `fromJson`) are the three factory methods on `GoogleClientSecret`, so those are all the ways to construct it. Happy to add a note making that explicit if it still feels unclear.

Reviewer: Did you push the intro change? Can't see any change.

Author: Yes check am

      'client_id': 'your-client-id.apps.googleusercontent.com',
      'client_secret': 'your-client-secret',
      'redirect_uris': [

>         'http://localhost:8082',

Reviewer: Motivate this change

Author: Changed the redirect URI to match the setup guide. Serverpod handles the Google OAuth callback on the **web server** (port 8082), not the API server (port 8080). The old example had the wrong port and included a `/auth/google/callback` path that doesn't exist in Serverpod's OAuth handling.

Reviewer: Great!

:::tip
You can also set these environment variables in your IDE's run configuration or CI/CD pipeline to avoid passing them manually each time.
:::

## GoogleIdpConfig parameter reference

| Parameter | Type | Required | Description |
| --- | --- | --- | --- |
| `clientSecret` | `GoogleClientSecret` | Yes | The Google OAuth client secret loaded from JSON. Can be loaded via `fromJsonString`, `fromJsonFile`, or `fromJson`. |
| `googleAccountDetailsValidation` | `Function?` | No | Custom validation callback for Google account details before allowing sign-in. Throws an exception to reject the account. |
| `getExtraGoogleInfoCallback` | `Function?` | No | Callback that receives the access token after sign-in, allowing you to call additional Google APIs and store extra user data. |
| `onAfterGoogleAccountCreated` | `Function?` | No | Callback invoked after a new Google account is created and linked to an auth user. Only called for new accounts, not returning users. |

> | `onBeforeAuthUserCreated` | `Function?` | No | Callback invoked before the auth user is created. Use this to set scopes or other data based on provider account info. |

Reviewer: There are no good code examples above on how to use the onBeforeAuthUserCreated. Maybe we should add one?

Author: Hmmm i think this is why references exist in the code we can link the reference instead here, these hooks can change and we don't want to update the docs anytime they change

Reviewer: Ok. Yeah, I guess what's unclear is what params the callback takes. Seems like Gemini knows how to use the hook.

There is also an 'onAfterAuthUserCreated'. Maybe we should add it? https://pub.dev/documentation/serverpod_auth_idp_server/latest/core/AuthUsersConfig/onAfterAuthUserCreated.html

Author: hmmm sadly it uses a type def and that doest get a reference 
<img width="593" height="357" alt="image" src="https://github.com/user-attachments/assets/57c5422f-d6c9-49d6-8e0b-04b7ba0fa306" />

Reviewer: Let's add two things:

1. Add 'onAfterAuthUserCreated' to the list. 
2. Add one code example using one of the hooks.

Author: Added
